# Notebook to organize my ABCD data analysis

## Variables of interest
### General 
* pds_ht2_y = Would you say that your growth in height?
* pds_skin2_y = Have you noticed any skin changes, especially pimples?
* pds_bdyhair_y = And how about the growth of your body hair? ("Body hair" means hair any place other than your head, such as under your arms) Would you say that your body hair growth:
### Female specific
* pds_f4_2_y = Have you noticed that your breasts have begun to grow?
* pds_f5_y = Have you begun to menstruate (started to have your period)?
### Male specific
* pds_m4_y = Have you noticed a deepening of your voice?
* pds_m5_y = Have you begun to grow hair on your face?

### Scoring Algorithms: 

For Items 1 through 4 on the girls’ version and all items on the boys’ version, response options were: not yet started (1 point); barely started (2 points); definitely started (3 points); seems complete (4 points); I don’t know (missing). Yes on the menstruation item = 4 points; no = 1 point. Point values are averaged for all items to give a Pubertal Development Scale (PDS) score. 

Puberty Category Scores are computed using the criteria of Crockett (1988, unpublished) by totaling the scale values given above. 

### To compute Puberty Category Scores for boys use body hair growth, voice change, and facial hair growth as follows:
* Prepubertal = 3
* Early Pubertal = 4 or 5 (no 3-point responses)
* Midpubertal = 6, 7, or 8 (no 4-points)
* Late pubertal = 9-11
* Postpubertal = 12


### To compute Puberty Category Scores for girls use body hair growth, breast development, and menarche as follows:
* Prepubertal = 2 and no menarche
* Early Puberty = 3 and no menarche
* Midpubertal = > 3 and no menarche
* Late Puberty = <= 7 and menarche
* Postpubertal = 8 and menarche.

In [1]:
library(psych)
library(reshape)
library(ggplot2)
library(plyr)
library(dplyr)
#install.packages("AGD")
library(AGD)


Attaching package: ‘ggplot2’

The following objects are masked from ‘package:psych’:

    %+%, alpha


Attaching package: ‘plyr’

The following objects are masked from ‘package:reshape’:

    rename, round_any


Attaching package: ‘dplyr’

The following objects are masked from ‘package:plyr’:

    arrange, count, desc, failwith, id, mutate, rename, summarise,
    summarize

The following object is masked from ‘package:reshape’:

    rename

The following objects are masked from ‘package:stats’:

    filter, lag

The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union

Warning message:
“package ‘AGD’ was built under R version 3.4.4”

### Important functions

In [2]:
BMI_cal<-function(height, weight){
  weight_kg<-(weight/2.20462)
  height_m<-(height*2.54)/100
  BMI=weight_kg/(height_m)^2
  return(BMI)
}

In [3]:
BMItiler<-function(data){
    datM<-subset(data, data$gender == "M")
    datM$interview_age<-(datM$interview_age)/12
    datM$zTest <- y2z(y=datM$BMI, x=datM$interview_age, sex="M", ref=cdc.bmi)
    datM$BMItile<-100*pnorm(datM$zTest)
    
    datF<-subset(data, data$gender == "F")
    datF$interview_age<-(datF$interview_age)/12
    datF$zTest <- y2z(y=datF$BMI, x=datF$interview_age, sex="F", ref=cdc.bmi)
    datF$BMItile<-100*pnorm(datF$zTest)
    
    data<-rbind.fill(datF, datM)
    return(data)
}

In [4]:
OVOB_kid<-function(BMItile, data){
  data$ov_ob[BMItile<=5]<-"Underweight"
  data$ov_ob[BMItile>5 & BMItile<=85]<-"Normalweight"
  data$ov_ob[BMItile>85 & BMItile<=95]<-"Overweight"
  data$ov_ob[BMItile>95]<-"Obese"
  return(data$ov_ob)
}

In [5]:
PCS<-function(x){
    earlyM <-subset(x, x$gender == "M")
    earlyM$sum <- earlyM$pds_m4_y  + earlyM$pds_m5_y + earlyM$pds_bdyhair_y

    earlyM$PCS[earlyM$sum <= 3]<-"prepubertal"
    earlyM$PCS[earlyM$sum > 3 & earlyM$sum <=5]<-"earlypubertal"
    earlyM$PCS[earlyM$sum > 5 & earlyM$sum <=8]<-"midpubertal"
    earlyM$PCS[earlyM$sum > 8 & earlyM$sum <=11]<-"latepubertal"
    earlyM$PCS[ earlyM$sum > 11]<-"latepubertal"
#    earlyM$PCS[earlyM$sum > 11]<-"latepubertal"

    early <-subset(x, x$pds_f5_y <= 1)

    early$sum <- early$pds_f4_2_y + early$pds_f5_y + early$pds_bdyhair_y
    early$PCS[early$sum == 2] <- "prepubertal"
    early$PCS[early$sum == 3] <- "earlypubertal"
    early$PCS[early$sum > 3] <- "midpubertal"

    late <-subset(x, x$pds_f5_y > 1)

    late$sum <- late$pds_f4_2_y + late$pds_f5_y + late$pds_bdyhair_y
    late$PCS[late$sum >= 7] <- "latepubertal"
#    late$PCS[late$sum > 7] <- "postpubertal"

    data<-rbind.fill(earlyM, early, late)
    return(data)
}

## Load data

In [6]:
anthro<-read.table("~/Google Drive/ABCD/important_txt/abcd_ypdms01.csv", sep=",", header=T)
puberty<-read.table("~/Google Drive/ABCD/important_txt/abcd_ant01.csv", sep=",", header=T)

## Check total data

In [7]:
x<-join(anthro, puberty)
dim(x)
x[x == 999] <- NA
dim(x)

Joining by: src_subject_id, interview_date, interview_age, gender, eventname


[1] 4524   34

[1] 4524   34

Have 34 vairables with 4524 datapoints
Now checking that they are all unique

In [8]:
duplicated(x$src_subject_id)

[1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [13] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [25] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [37] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [49] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [61] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [73] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [85] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
  [97] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [109] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [121] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [133] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [145] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [157] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [169] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [181] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [193] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [205] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [217] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [229] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [241] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [253] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [265] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [277] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [289] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [301] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [313] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [325] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [337] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [349] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [361] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [373] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [385] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [397] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [409] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [421] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [433] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [445] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [457] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [469] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [481] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [493] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [505] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [517] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [529] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [541] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [553] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [565] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [577] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [589] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE
 [601] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALS

### Calculate BMI

In [9]:
x$BMI<-mapply(BMI_cal, height=x$anthroheightcalc, weight=x$anthroweightcalc)

In [10]:
y<-BMItiler(data = x)

In [11]:
data<-PCS(y)

In [12]:
data$OVOB<-OVOB_kid(BMItile = data$BMItile, data = data)

### Missing puberty data

In [13]:
dataMissing<-data[is.na(data$PCS),]

In [14]:
dim(dataMissing)

[1] 387  40

In [15]:
summary(dataMissing)

             src_subject_id interview_date interview_age    gender 
 sub-NDARINVANUGPBR4:  2    3/25/17:  6    Min.   : 8.917   F:378  
 sub-NDARINVHBZ05H20:  2    6/19/17:  6    1st Qu.: 9.250   M:  9  
 sub-NDARINV03XVEBPM:  1    5/13/17:  5    Median : 9.750          
 sub-NDARINV042UJKFB:  1    6/24/17:  5    Mean   : 9.812          
 sub-NDARINV0C471G23:  1    6/26/17:  5    3rd Qu.:10.333          
 sub-NDARINV0GND16RW:  1    6/28/17:  5    Max.   :11.000          
 (Other)            :379    (Other):355                            
                 eventname     pds_sex_y   pds_ht2_y      pds_skin2_y   
 baseline_year_1_arm_1:387   Min.   :2   Min.   :1.000   Min.   :1.000  
                             1st Qu.:2   1st Qu.:2.000   1st Qu.:1.000  
                             Median :2   Median :2.000   Median :1.000  
                             Mean   :2   Mean   :2.301   Mean   :1.509  
                             3rd Qu.:2   3rd Qu.:3.000   3rd Qu.:2.000  
                  

### Descriptives of the data 

In [16]:
mytable <- xtabs(~PCS+OVOB+gender, data=data)
ftable(mytable) # print table 
summary(mytable) # chi-square test of indepedence

                           gender   F   M
PCS           OVOB                       
earlypubertal Normalweight        348 729
              Obese                33 184
              Overweight           32 158
              Underweight          40  44
latepubertal  Normalweight         24  24
              Obese                18  10
              Overweight           13   9
              Underweight           0   0
midpubertal   Normalweight        824 321
              Obese               187  92
              Overweight          193  83
              Underweight          47  12
prepubertal   Normalweight          0 485
              Obese                 0 105
              Overweight            0  84
              Underweight           0  25

Call: xtabs(formula = ~PCS + OVOB + gender, data = data)
Number of cases in table: 4124 
Number of factors: 3 
Test for independence of all factors:
	Chisq = 1325.1, df = 24, p-value = 4.912e-265
	Chi-squared approximation may be incorrect

In [17]:
summary(as.factor(data$PCS))

earlypubertal  latepubertal   midpubertal   prepubertal          NA's 
         1569            98          1759           699           387

In [18]:
summary(as.factor(data$OVOB))

Normalweight        Obese   Overweight  Underweight         NA's 
        2995          690          644          181            2

### Removing the subject who are prepubertal and/or underweight

In [19]:
DF<-subset(data, data$PCS != "prepubertal")
DF<-subset(DF, DF$OVOB !="Underweight")

In [20]:
mytable <- xtabs(~PCS+OVOB+gender, data=DF)
ftable(mytable) # print table 
summary(mytable)

                           gender   F   M
PCS           OVOB                       
earlypubertal Normalweight        348 729
              Obese                33 184
              Overweight           32 158
latepubertal  Normalweight         24  24
              Obese                18  10
              Overweight           13   9
midpubertal   Normalweight        824 321
              Obese               187  92
              Overweight          193  83

Call: xtabs(formula = ~PCS + OVOB + gender, data = DF)
Number of cases in table: 3282 
Number of factors: 3 
Test for independence of all factors:
	Chisq = 640.3, df = 12, p-value = 2.577e-129

### Limiting factor is the late pubertal with a cell of 9. 

## Who has scans

In [21]:
avail_scans <- read.table("~/Google Drive/ABCD/important_txt/scan_list.txt")

In [22]:
avail_scans$have_data <- rep(1,nrow(avail_scans))

In [23]:
names(avail_scans)<-c("src_subject_id", "have_data")

In [24]:
dim(avail_scans)

[1] 1568    2

### Combining the scanning and demographic data

In [25]:
data0<-join(DF, avail_scans)

Joining by: src_subject_id


In [26]:
dim(data0)

[1] 3282   41

In [27]:
data<-data0[!is.na(data0$have_data),]

In [28]:
dim(data)

[1] 1149   41

In [29]:
mytable <- xtabs(~gender+PCS+OVOB, data=data)
ftable(mytable) # print table 
summary(mytable) # chi-square test of indepedence

                     OVOB Normalweight Obese Overweight
gender PCS                                             
F      earlypubertal               130     8          2
       latepubertal                 20    11         11
       midpubertal                 271    68         60
M      earlypubertal               240    56         55
       latepubertal                 17     8          8
       midpubertal                 122    34         28

Call: xtabs(formula = ~gender + PCS + OVOB, data = data)
Number of cases in table: 1149 
Number of factors: 3 
Test for independence of all factors:
	Chisq = 209.12, df = 12, p-value = 4.248e-38

## Get a list of subjects needed for analysis

In [30]:
randomly<-function(y){
    nines<-y[sample(nrow(y), 2), ]
    return(nines)
}

In [31]:
#randomly(data)

### Function to find and label by pubertal status, weight, and sex

In [32]:
get_subs<-function(x){
    big_list<-list(
        ob_late_F<-subset(x, x$OVOB == "Obese" & x$PCS == "latepubertal" & x$gender == "F"),
        ob_mid_F<-subset(x, x$OVOB == "Obese" & x$PCS == "midpubertal" & x$gender == "F"),
         ob_early_F<-subset(x, x$OVOB == "Obese" & x$PCS == "earlypubertal" & x$gender == "F"),
    
        ov_late_F<-subset(x, x$OVOB == "Overweight" & x$PCS == "latepubertal" & x$gender == "F"),
        ov_mid_F<-subset(x, x$OVOB == "Overweight" & x$PCS == "midpubertal" & x$gender == "F"),
         ov_early_F<-subset(x, x$OVOB == "Overweight" & x$PCS == "earlypubertal" & x$gender == "F"),
    
        no_late_F<-subset(x, x$OVOB == "Normalweight" & x$PCS == "latepubertal" & x$gender == "F"),
        no_mid_F<-subset(x, x$OVOB == "Normalweight" & x$PCS == "midpubertal" & x$gender == "F"),
        no_early_F<-subset(x, x$OVOB == "Normalweight" & x$PCS == "earlypubertal" & x$gender == "F"),
    ################################################################################################
        ob_late_M<-subset(x, x$OVOB == "Obese" & x$PCS == "latepubertal" & x$gender == "M"),
        
        ob_mid_M<-subset(x, x$OVOB == "Obese" & x$PCS == "midpubertal" & x$gender == "M"),
        ob_early_M<-subset(x, x$OVOB == "Obese" & x$PCS == "earlypubertal" & x$gender == "M"),

        ov_late_M<-subset(x, x$OVOB == "Overweight" & x$PCS == "latepubertal" & x$gender == "M"),
        
        ov_mid_M<-subset(x, x$OVOB == "Overweight" & x$PCS == "midpubertal" & x$gender == "M"),
        ov_early_M<-subset(x, x$OVOB == "Overweight" & x$PCS == "earlypubertal" & x$gender == "M"),

        no_late_M<-subset(x, x$OVOB == "Normalweight" & x$PCS == "latepubertal" & x$gender == "M"),
        
        no_mid_M<-subset(x, x$OVOB == "Normalweight" & x$PCS == "midpubertal" & x$gender == "M"),
        no_early_M<-subset(x, x$OVOB == "Normalweight" & x$PCS == "earlypubertal" & x$gender == "M")
    )
#     check_dims<-lapply(big_list, dim)
#    final_list<-lapply(big_list, randomly)
    return(big_list)
}

In [33]:
data2<-get_subs(data)

In [34]:
length(data2)

[1] 18

In [35]:
x<-ldply(data2, rbind)

In [36]:
dim(x)

[1] 1149   41

In [38]:
head(x)

src_subject_id      interview_date interview_age gender eventname            
1 sub-NDARINVULZMADF1 3/20/17        11.00000      F      baseline_year_1_arm_1
2 sub-NDARINV0W4WA8RV 5/16/17        11.00000      F      baseline_year_1_arm_1
3 sub-NDARINV8PHEF3UH 12/15/16       10.75000      F      baseline_year_1_arm_1
4 sub-NDARINV4D917H14 8/14/17        10.91667      F      baseline_year_1_arm_1
5 sub-NDARINVT472A3DT 6/5/17         10.83333      F      baseline_year_1_arm_1
6 sub-NDARINVMFVA1BZF 3/22/17        10.25000      F      baseline_year_1_arm_1
  pds_sex_y pds_ht2_y pds_skin2_y pds_bdyhair_y pds_f4_2_y ⋯ anthroweight3lb
1 2          3        2           3             2          ⋯ NA             
2 2          4        2           4             2          ⋯ NA             
3 2         NA        4           4             4          ⋯ NA             
4 2         NA        2           4             4          ⋯ NA             
5 2          4        3           2             3          ⋯ NA             
6 2          2        2           2             2          ⋯ NA             
  anthroweightcalc anthro_waist_cm BMI      zTest BMItile  sum PCS         
1 189.0            43.0            33.47954 2.504 99.38601  9  latepubertal
2 159.0            39.0            29.08122 2.206 98.63080 10  latepubertal
3 141.0            35.5            29.46877 2.269 98.83658 12  latepubertal
4 197.0            39.5            34.34928 2.558 99.47362 12  latepubertal
5 151.0            32.0            25.12749 1.816 96.53148  9  latepubertal
6 115.8            35.0            26.96323 2.116 98.28276  8  latepubertal
  OVOB  have_data
1 Obese 1        
2 Obese 1        
3 Obese 1        
4 Obese 1        
5 Obese 1        
6 Obese 1

In [40]:
myvars<-c('src_subject_id','gender','PCS', 'OVOB', 'BMItile', 'sum', 'interview_age')
y<-x[myvars]

In [41]:
write.table(y, "~/Google Drive/ABCD/important_txt/data4analysis2.txt", sep=" ", col.names=F, row.names = F)

## Check missing data

In [37]:
missing<-data0[is.na(data0$have_data),]

In [38]:
dim(missing)

[1] 2133   41

In [39]:
mytable <- xtabs(~gender+PCS+OVOB, data=missing)
ftable(mytable) # print table 
summary(mytable) # chi-square test of indepedence

                     OVOB Normalweight Obese Overweight
gender PCS                                             
F      earlypubertal               218    25         30
       latepubertal                  4     7          2
       midpubertal                 553   119        133
M      earlypubertal               489   128        103
       latepubertal                  7     2          1
       midpubertal                 199    58         55

Call: xtabs(formula = ~gender + PCS + OVOB, data = missing)
Number of cases in table: 2133 
Number of factors: 3 
Test for independence of all factors:
	Chisq = 450.5, df = 12, p-value = 7.232e-89
	Chi-squared approximation may be incorrect

Looks like the cell sizes are really uneven, will need to increase data in the female scans in particular

In [40]:
dim(missing[missing$OVOB == "Obese",])

[1] 339  41

In [41]:
ov_early<-subset(missing, missing$OVOB == "Overweight" & missing$PCS == "earlypubertal" & missing$gender == "F")
dim(ov_early)

[1] 30 41

In [42]:
ob_late<-subset(missing, missing$OVOB == "Obese" & missing$PCS == "latepubertal")
dim(ob_late)

[1]  9 41

In [43]:
ob_post<-subset(missing, missing$OVOB == "Obese" & missing$PCS == "postpubertal")
dim(ob_post)

[1]  0 41

In [44]:
ov_post<-subset(missing, missing$OVOB == "Overweight" & missing$PCS == "postpubertal")
dim(ov_post)

[1]  0 41

In [45]:
ov_late<-subset(missing, missing$OVOB == "Overweight" & missing$PCS == "latepubertal")
dim(ov_late)

[1]  3 41

In [46]:
no_late<-subset(missing, missing$OVOB == "Normalweight" & missing$PCS == "latepubertal")
dim(no_late)

[1] 11 41

In [47]:
no_post<-subset(missing, missing$OVOB == "Normalweight" & missing$PCS == "postpubertal")
dim(no_post)

[1]  0 41

In [48]:
gather<-rbind.fill(ob_late, ob_post, ov_post, ov_late, no_late, no_post)
dim(gather)

[1] 23 41

In [49]:
mytable <- xtabs(~gender+PCS+OVOB, data=gather)
ftable(mytable) # print table 
summary(mytable)

                    OVOB Normalweight Obese Overweight
gender PCS                                            
F      latepubertal                 4     7          2
M      latepubertal                 7     2          1

Call: xtabs(formula = ~gender + PCS + OVOB, data = gather)
Number of cases in table: 23 
Number of factors: 3 
Test for independence of all factors:
	Chisq = 3.599, df = 2, p-value = 0.1654
	Chi-squared approximation may be incorrect

In [50]:
#write.table(gather$src_subject_id, "~/Google Drive/ABCD/important_txt/missing_grab.txt", sep="\t", row.names=F)

In [51]:
#write.table(ov_early$src_subject_id, "~/Google Drive/ABCD/important_txt/missing_grab2.txt", sep="\t", row.names=F)